armory/scenarios/poison.py

```diff
@@ -386,6 +390,46 @@ def finalize_results(self):
             log.info(
                 f"Test targeted misclassification accuracy: {self.poisoned_targeted_test_metric.mean():.2%}"
             )
+        
+        self.results["N_poisoned_data"] = int(len(self.poison_index))
+
+        if self.use_filtering_defense:
+           
+            filtered = (1 - self.indices_to_keep).astype(np.bool)
+            poisoned = np.zeros_like(self.y_clean).astype(np.bool)
+            poisoned[self.poison_index] = True
+
+            false_negatives = int(np.sum(~filtered & poisoned))
+            true_positives = int(np.sum(filtered & poisoned))
+            true_negatives = int(np.sum(~filtered & ~poisoned))
+            false_positives = int(np.sum(filtered & ~poisoned))
+            
+            false_negative_rate = false_negatives / np.sum(poisoned)
+            true_positive_rate = true_positives / np.sum(poisoned)
+            true_negative_rate = true_negatives / np.sum(~poisoned)
+            false_positive_rate = false_positives / np.sum(~poisoned)
+            
+            f1_score = true_positives / (true_positives + .5 * (false_positives + false_negatives))
+            
+            self.results["filter_true_positives"] = true_positives
+            self.results["filter_false_positives"] = false_positives
+            self.results["filter_true_negatives"] = true_negatives
+            self.results["filter_false_negatives"] = false_negatives
+            self.results["filter_true_positive_rate"] = true_positive_rate
+            self.results["filter_false_positive_rate"] = false_positive_rate
+            self.results["filter_true_negative_rate"] = true_negative_rate
+            self.results["filter_false_negative_rate"] = false_negative_rate
+            self.results["filter_f1_score"] = f1_score
+
+            self.results["fraction_data_removed_by_filter"] = filtered.mean()
```

Comment:
cosmetics, but I might change `"fraction_data_removed_by_filter"` to `"filter_fraction_data_removed"` and similarly for `"N_datapoints_removed_by_filter"` so that all `filter` related metrics start with `filter_` and thus appear side-by-side in the output JSON

Reply:
good idea